### PR TITLE
file preview panel

### PR DESCRIPTION
**Pull request by leangseu-edx**
Review instruction:
- filter out samples files. I will remove those before merge.
- use [this pr](https://github.com/edx/edx-platform/pull/29491) on edx-platform

There are additional support can be done such as mp4 files, although I am not sure if that is necessary.

### PDF View
Screen Shot 2021-12-01 at 10 44 28 AM
### Image/Text
Screen Shot 2021-12-01 at 10 44 11 AM
### Unsupported
Screen Shot 2021-12-01 at 10 43 53 AM
### No attache files
Screen Shot 2021-12-01 at 10 56 11 AM

### CODE REVIEW

File: src/components/PreviewPanel/index.test.jsx [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+
+  let el;
+  beforeEach(() => {
+    el = shallow(<PreviewPanel {...props} />);
+  });
+  describe('snapshot', () => {
+    test('default', () => {
+      expect(el).toMatchSnapshot();
+    });
+    test('PDFRenderer', () => {
```
Comment (leangseu-edx) [inline src/components/PreviewPanel/index.test.jsx:29]: Questionable whether to test this files.

File: src/components/PreviewPanel/PDFRenderer.jsx [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+          </div>
+        </Document>
+        <ActionRow className="d-flex justify-content-center m-0">
+          <IconButton
+            size="inline"
+            alt="previous pdf page"
+            iconAs={Icon}
+            src={ChevronLeft}
+            disabled={!this.hasPrev}
+            onClick={() => this.setPageNumber(this.state.pageNumber - 1)}
```
Comment (muselesscreator) [inline src/components/PreviewPanel/PDFRenderer.jsx]: onClick and onChange should be methods :-(

File: src/components/PreviewPanel/PDFRenderer.test.jsx [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+    el.instance().onDocumentLoadSuccess = jest
+      .fn()
+      .mockName('onDocumentLoadSuccess');
+    el.instance().onDocumentLoadError = jest
+      .fn()
+      .mockName('onDocumentLoadError');
+    el.instance().onLoadPageSuccess = jest.fn().mockName('onLoadPageSuccess');
+  });
+  test('snapshot', () => {
+    expect(el).toMatchSnapshot();
```
Comment (muselesscreator) [inline src/components/PreviewPanel/PDFRenderer.test.jsx]:
```suggestion
    expect(el.instance().render()).toMatchSnapshot();
```

File: src/components/PreviewPanel/index.jsx [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+  static supportedTypes = Object.keys(PreviewPanel.renderers);
+
+  static isSupported = (fileName) => {
+    const fileType = fileName.split('.').pop();
+    return PreviewPanel.supportedTypes.includes(fileType);
+  };
+
+  get fileType() {
+    if (this.props.fileName) {
+      const regex = /(?:\.([^.]+))?$/;
```
Comment (muselesscreator) [inline src/components/PreviewPanel/index.jsx]: regex need explanations on what they are doing.  why are there 2 ways of checking the filetype?

File: src/components/PreviewPanel/index.jsx [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+
+const DefaultRenderer = () => <div>Unsupported files</div>;
+
+const defaultPlugins = [PDFRenderer, ImageRenderer];
+
+/**
+ * <PreviewPanel />
+ */
+export class PreviewPanel extends React.Component {
+  static renderers = defaultPlugins.reduce(
```
Comment (muselesscreator) [inline src/components/PreviewPanel/index.jsx:16]: potentially simpler counter:
```js
get supportedRenderer() {
   const renderers = defaultPlugins.filter(({ suppotedTypes }) => supportedTypes.includes(this.fileType));
   return renderers.length ? renderers[0] : null;
}
get renderer() {
  const supported = this.supportedRenderer;
  return supported ? supported : DefaultRenderer;
}   
get isSupported() {
  this.supportedRenderer !== null;
}
```

**Pull request closed without merging**